pkg/s3select/json/reader.go

```diff
@@ -135,21 +161,7 @@ func (or *objectReader) Read(p []byte) (n int, err error) {
 		}
 	}
 
-	index := or.objectEndIndex(p, n)
-	if index == -1 || index == n {
-		return n, nil
-	}
-
-	or.endOfObject = true
-	if or.p == nil {
-		or.p = p
-		or.start = index
-		or.end = n
-	} else {
-		or.start -= index
-	}
-
-	return index, nil
+	return or.updateState(p[:n]), nil
```

Comment:
Playing with trimming may result in new slice creation (not for the end index, surely for start index, better it can be avoided).  That's the reason I didn't use this style previously.

Reply:
Slicing a slice doesn't copy the slice https://blog.golang.org/go-slices-usage-and-internals it's quite common in Go so that's why I added this style, there is no reason to take `n` as an argument.

Reply:
I would love to keep a proper name for it, my idea was to document the portion which has some intricate logic of maintaining offsets.

So suggest a good name though.